Update gateway antenna location only when auth is available (#3326)
TheThingsNetwork/lorawan-stack: CHANGELOG.md

- Improve LinkADRReq scheduling condition computation and, as a consequence, downlink task efficiency.
- CUPS Server only accepts The Things Stack API Key for token auth.
- Improve MQTT Pub/Sub task restart conditions and error propagation.

> - GS is now consistent in automatically updating gateway locations only if the gateway is connected via an authenticated frontend (gRPC, MQTT, or BasicStation).

Reviewer: ```suggestion
- Gateway Server is now updates the gateway location only if the gateway is authenticated.
```
